### PR TITLE
BP-802 Add verification method iDIN (age verification)

### PR DESCRIPTION
**Pull request by saranchuk-hys**

### CODE REVIEW

File: Controller/Checkout/Idin.php [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+    /**
+     * @var Log
+     */
+    protected $logger;
+
+    /**
+     * @var \Magento\Framework\HTTP\Client\Curl
+     */
+    protected $curlClient;
```
Comment (vladislav-padalka-hys) [inline Controller/Checkout/Idin.php:85]: is not in use

File: Controller/Checkout/Idin.php [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+    /**
+     * @var Account
+     */
+    protected $configProviderAccount;
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManager;
```
Comment (vladislav-padalka-hys) [inline Controller/Checkout/Idin.php:95]: is not in use

File: Controller/Checkout/Idin.php [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    /** @var CheckoutSession */
+    protected $_checkoutSession;
+
+    /** @var Encryptor $encryptor */
+    private $encryptor;
```
Comment (vladislav-padalka-hys) [inline Controller/Checkout/Idin.php:101]: is not in use

File: Gateway/Http/TransactionBuilder/DataRequest.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ */
+
+namespace Buckaroo\Magento2\Gateway\Http\TransactionBuilder;
+
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Data\Form\FormKey;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\Encryption\Encryptor;
+use Buckaroo\Magento2\Gateway\Http\Transaction;
+use Buckaroo\Magento2\Model\ConfigProvider\Account;
```
Comment (vladislav-padalka-hys) [inline Gateway/Http/TransactionBuilder/DataRequest.php:29]: not in use

File: Model/Config/Source/Idinmode.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * Options getter
+     *
+     * @return array
+     */
+    public function toOptionArray()
+    {
+        return [
+            ['value' => 0, 'label' => __('Global')],
+            ['value' => 1, 'label' => __('Per Product')],
```
Comment (vladislav-padalka-hys) [inline Model/Config/Source/Idinmode.php:33]: constants usage ?

File: view/frontend/web/js/view/checkout/idin/idin-step-view.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+        verificateIDIN: function () {
+            var el = document.getElementById("buckaroo_magento2_idin_issuer");
+            var issuer = el.options[el.selectedIndex].value;
+                self = this;
+
+                if(!issuer){
+                    alert({
+                        title: $t('Error'),
+                        content: $t('Please choose bank'),
```
Comment (vladislav-padalka-hys) [inline view/frontend/web/js/view/checkout/idin/idin-step-view.js:76]: please adjust texts with what I've gave to you

**Comment by vladislav-padalka-hys** [comment]
@saranchuk-hys what will be in case of guest customer ? do you support such case ? will verification be stored in session ?

**Comment by vladislav-padalka-hys** [comment]
I will post further workflow and UI questions/notes to Jira

File: Controller/Checkout/Idin.php (hunk 1 above)
Comment (saranchuk-hys) [reply Controller/Checkout/Idin.php]: fixed

File: Controller/Checkout/Idin.php (hunk 3 above)
Comment (saranchuk-hys) [reply Controller/Checkout/Idin.php]: fixed

File: Model/Config/Source/Idinmode.php (hunk 5 above)
Comment (saranchuk-hys) [reply Model/Config/Source/Idinmode.php]: fixed

File: view/frontend/web/js/view/checkout/idin/idin-step-view.js (hunk 6 above)
Comment (saranchuk-hys) [reply view/frontend/web/js/view/checkout/idin/idin-step-view.js]: fixed

File: Controller/Checkout/Idin.php (hunk 2 above)
Comment (saranchuk-hys) [reply Controller/Checkout/Idin.php:89]: fixed

File: Gateway/Http/TransactionBuilder/DataRequest.php (hunk 4 above)
Comment (saranchuk-hys) [reply Gateway/Http/TransactionBuilder/DataRequest.php]: fixed

File: Model/ConfigProvider/Idin.php [hunk 7]
```diff
@@ -138,6 +142,10 @@ protected function isIDINActive()
                 $customer              = $this->customerRepositoryInterface->getById($customerId);
                 $customerAttributeData = $customer->__toArray();
                 $active                = (isset($customerAttributeData['custom_attributes']) && isset($customerAttributeData['custom_attributes']['buckaroo_idin']) && $customerAttributeData['custom_attributes']['buckaroo_idin']['value']) ? false : 1;
+            }elseif($active === true){
```
Comment (vladislav-padalka-hys) [inline Model/ConfigProvider/Idin.php:145]: minor thing, but could you please follow phpcs to eliminate such things (e.g.: wrong space indentation)

Comment (saranchuk-hys) [reply Model/ConfigProvider/Idin.php]: sorry, fixed

**Review by vladislav-padalka-hys — APPROVED** [review]

**Pull request merged**